firefox-ios/Client/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -657,7 +687,17 @@ class SearchViewController: SiteTableViewController,
             let suggestion = viewModel.historySites[indexPath.item]
             searchDelegate?.searchViewController(self, didHighlightText: suggestion.url, search: false)
         case .searchSuggestions:
-            guard let suggestion = viewModel.suggestions?[indexPath.item] else { return }
+            // Ecosia: Check if this is the AI Search item
+            if isAISearchRow(indexPath) {
+                handleAISearchHighlight(indexPath)
+                return
+            }
+
+            /* Ecosia: Modified to handle AI Search item and use safe array access
+            let suggestion = viewModel.suggestions?[indexPath.item] ?? ""
+            searchDelegate?.searchViewController(self, didHighlightText: suggestion, search: false)
```

Comment:
I see this line is duplicated in and out of the comment, probably can be simplified.